### ISSUE TITLE
Two lists in a single controller, and both of the lists of models don't have ids

### ISSUE THREAD

**Issue by ColtonIdle** [airbnb/epoxy#811]
I'm modeling some data right now, and up to this point every time I've used epoxy the models have had some id from my backend. At this point though, the two lists I'm generating don't come from the backend so I generated these models on the Android side. I loop through each list in the controller and add it to the list. The thing is, I've just done

```
listOne.forEachIndexed{ index, it ->
id(index)
myItem(it)
}

listTwo.forEachIndexed{ index, it ->
id(index)
myItem(it)
}
```

This seems to work, but I have two questions.
1. Is it alright to pass an index like this as an id? Or should my kotlin data class that are modeling my views (not epoxy models, just my models of data in general) have some kind of id added to them?
2. At this point, I'll have two items with an id of 0. Two items in the same controller with the id of 1. Etc, etc. Is this bad?

**Comment by elihart** [comment]
are these lists being used in carousel? There is a strict rule that models cannot have the same id, so you can't add models to the same controller that both have `0` as the id. Make sure you have debug logging enabled to alert you to these issues.

Using index as an id can be ok in situations where you know that items won't be added or removed from the list - ie you know you have static content. Otherwise if the list structure changes animations will not work correctly.

If you want to use index for two lists then you should name space then.
```
listOne.forEachIndexed{ index, it ->
id("list one", index)
myItem(it)
}
```

**Comment by ColtonIdle** [comment]
1. Can you expand on the debug logging? I read about it in the docs, but it didn't quite make sense. It seems like one of the debug logs is on by default?

2. Alright. I will namespace them. Thank you for the heads up. If I think the content could change and so it's not static, do you think I should just generate some kind of UUIDs locally then?

**Comment by elihart** [comment]
This is the logging I was thinking of 
https://github.com/airbnb/epoxy/wiki/Epoxy-Controller#filtering-duplicates

You can also use `setGlobalExceptionHandler`, and `setGlobalDuplicateFilteringDefault` on `EpoxyController` to easily set default handling for debug builds - this is important for getting alerted to issues.

You can use a UUID, as long as you can keep the association between UUID and item. To be clear, it is ok if a single item's content changes - it will have the same index in the list and therefore the same id. You could also add new items to the end of the list, since those would have new indices and new ids.

You can't remove an item, or insert a new item, in the start or middle of the list though, since the RecyclerView will see the index as the same id and therefore think that an item has been updated, not that one has been inserted or removed - so you will get a change animation instead of a remove/insert animation.

**Comment by ColtonIdle** [comment]
Okay. Thanks for the clarity. I can keep the item id, BUT I would need a new instance of that item right? Just because my object is mutable and the id is the same, but I change a property, Epoxy won't update it right?

**Comment by elihart** [comment]
yes, if your `equals` function of the object reports that the two objects are equal (which it will for the same instance), that mutating an object and reusing it will not trigger an update.

Immutable kotlin data classes work best (use the copy function)
